### ISSUE TITLE
Support Custom Browsertrix Behaviors Loading

### ISSUE THREAD

**Issue by ikreymer** [webrecorder/browsertrix-crawler#263]
With switch of Browsertrix Behaviors to inheritance-less model, it should be easier to support custom behaviors in Browsertrix Crawler!

Possible CLI:
`--behaviors /path/to/my-behavior.js --behaviors /path/to/behaviors-dir`

The behavior, eg. `my-behavior.js` would contain a class like:

```
class MyBehavior
{
   static isMatch() {
    return window.location.href.match(/mypagerx/)
  }
  
  async* run(ctx) {
    ...
  }
}
```

While we currently support playwright (previously) puppeteer drivers, this would provide a perhaps simpler direct way to create behaviors, which can also be used in ArchiveWeb.page

Later, behaviors may also be expanded to support customizations to link extraction and crawling, for example by adding a `ctx.addLink(url)` or `ctx.extractLinks("a[selector]")` calls that would queue up additional URLs.

**Comment by edsu** [comment]
I think this is going to be awesome!

**Comment by ikreymer** [comment]
Merged via #285 and released in 0.10.2!
